File: sams-backend/src/main/java/com/sams/entity/Priority.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_priority", schema = "purchase")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Priority {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "priority_id")
    private Long priorityId;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "priority_name")
    private String priorityName;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "org_name")
    private String orgName;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "location_name")
    private String locationName;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}

File: SAMSUI/src/app/Model/master/priority.ts
import { BaseModel } from "../base/baseModel";

export class PriorityModel extends BaseModel {

    public priorityId ?: number;

    public orgId ?: number;

    public priorityName ?: string;

    public active ?: boolean;

    public createdBy ?:string;

    public createdDt ?: Date;

    public createdDtDisp ?: string;

    public updatedBy ?: string;

    public updatedDt ?: Date;

    public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public orgName ?: string;
    
    public locationId ?:number;

    public locationName ?:string;
}
